### PR TITLE
Add ability to pass plugin properties on invoice runs. See #1471

### PR DESCRIPTION
**Pull request by sbrossie**
Depends on https://github.com/killbill/killbill-api/pull/92

We also need regenerate the java client to complete the work, but the jaxrs apis are not broken as the plugin properties are passed as query parameters. The java client signatures would change though.

### CODE REVIEW

File: invoice/src/main/java/org/killbill/billing/invoice/InvoicePluginDispatcher.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ public DateTime priorCall(final LocalDate targetDate,
                               final boolean isRescheduled,
                               final CallContext callContext,
                               // The pluginProperties list passed to plugins is mutable by the plugins
-                              @SuppressWarnings("TypeMayBeWeakened") final LinkedList<PluginProperty> properties,
+                              @SuppressWarnings("TypeMayBeWeakened") final Iterable<PluginProperty> properties,
```
Comment (pierre) [inline invoice/src/main/java/org/killbill/billing/invoice/InvoicePluginDispatcher.java:97]: NACK. There's a reason why we have a `@SuppressWarnings("TypeMayBeWeakened")` here 😄 

See https://github.com/killbill/killbill/pull/1333/files

**Review by pierre — CHANGES_REQUESTED** [review]

File: invoice/src/main/java/org/killbill/billing/invoice/InvoicePluginDispatcher.java (hunk 1 above)
Comment (sbrossie) [reply invoice/src/main/java/org/killbill/billing/invoice/InvoicePluginDispatcher.java:97]: See https://github.com/killbill/killbill/pull/1696

Comment (sbrossie) [reply invoice/src/main/java/org/killbill/billing/invoice/InvoicePluginDispatcher.java]: PR has been merged and should address the comment.

File: pom.xml [hunk 2]
```diff
@@ -59,8 +59,10 @@
         <url>http://github.com/killbill/killbill</url>
     </issueManagement>
     <properties>
+        <check.fail-enforcer>false</check.fail-enforcer>
```
Comment (pierre) [inline pom.xml:62]: You can use parent pom `0.145.3-e8112a5-SNAPSHOT`.

**Review by pierre — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (sbrossie) [reply pom.xml]: Fixed in https://github.com/killbill/killbill/pull/1695/commits/bb4bf4dd1168409ec025644a41307fb43aab2f85

**Pull request merged**